### PR TITLE
mask plugin and permissions styles

### LINKED ISSUE #1374: Broken Permission styles and v-mask issue
# Bug Report

## Steps to Reproduce

1. Paste to CLI  `git clone git@github.com:directus/app.git fresh_app && cd fresh_app && yarn install && cp public/config_example.js public/config.js && yarn dev`
2. Login in to Demo API
3. Goto Admin Settings > Permissions > Public

## Expected Behavior
no erros and broken styles
## Actual Behavior
see https://github.com/directus/app/pull/1373
## Other Context & Screenshots
a quick fix is already provided here see https://github.com/directus/app/pull/1373

## Technical Details

- Device: [macbook]
- OS: [MacOS]
- Browser: [Chrome latest]
- Directus App: [latest
- Directus API: [latest]
- Install Method: [cloned `master` branch of `directus/app`]

### PR DESCRIPTION
**Pull request by dev7ch**
> Update: I did a fresh clone an install as well

This PR should solve several issues:

1. The new mask plugin isn`t really "futureproof" maintained and has an infinite loop, see https://github.com/vuejs-tips/vue-the-mask/pull/64 ... but i fixed it by a workaround (hopefully)

2. Somehow the permissions table styling is totally broken (locally and your demo api as well) ... i am not sure where this comes from, but i fixed it ... hope you are fine with it 😺 

image

3. add yarn lock to git ignore

4. Upgrade vue, was getting a mismatching error of vue and vue-template

closes #1374

### CODE REVIEW

**Comment by dev7ch** [comment]
and i didnt test if `v-mask` is working same as `:mask` ... please could you check this before or let me know where this module is in action

**Comment by rijkvanzanten** [comment]
`v-mask` supposedly works on any `<input>` element or any other content-editable, while `:mask` only works on the mask component. What is causing the infinite loop exactly? The default for mask should be `null`, which should not activate the `v-mask` attribute right?

**Comment by rijkvanzanten** [comment]
Also, the CSS of permissions looks fine on my end?

image

**Comment by dev7ch** [comment]
@rijkvanzanten i was almost sure i was too fast but a fresh install gives me the same result.

- node v.8.14.0
all done with yarn

After login: 

image


After create a section and go to Permissions:

image


Console:

image


the `mask` : `null` is not valid for "String" i think ... and the

**Comment by dev7ch** [comment]
> What is causing the infinite loop exactly? 
vuejs-tips/vue-the-mask#64 

> The default for mask should be null, which should not activate the v-mask attribute right?
I think `null` is not valid for type of `String` in vue

**Comment by dev7ch** [comment]
hey @rijkvanzanten did you have time to think about this? 
I think there are 2 possible options:

- continue for this release for with `vue-the-mask` and another mask plugin in the next release?
- try another plugin before release it

File: src/components/permissions/permissions.vue [hunk 1]
```diff
@@ -156,16 +156,17 @@ export default {
 };
 </script>
 
-<style lang="scss" scoped>
-.interface {
-  margin-bottom: 40px;
-}
+<style lang="scss">
```
[the remark this answers is not in the document]
Comment (rijkvanzanten) [reply src/components/permissions/permissions.vue]: Nope! That's too risky IMO. Instead we should replace the `>>>` with `/deep/` (revert that PR)

[the remark this answers is not in the document]
Comment (dev7ch) [reply src/components/permissions/permissions.vue]: done :)

File: package.json [hunk 2]
```diff
@@ -46,7 +46,7 @@
     "shortid": "^2.2.12",
     "snarkdown": "^1.2.2",
     "v-tooltip": "^2.0.0-rc.33",
-    "vue": "^2.6.0",
+    "vue": "^2.6.8",
```
[the remark this answers is not in the document]
Comment (rijkvanzanten) [reply package.json]: Not that I'm aware of. Patch updates shouldn't introduce any incompatibility issues, and I trust the Vue community to not publish 💣s

**Comment by rijkvanzanten** [comment]
I think we should merge and launch this workaround for vue-the-mask, and keep looking for a better alternative in the future.

**Comment by rijkvanzanten** [comment]
The workaround for vue-the-mask actually broke support for masking :(

**Comment by rijkvanzanten** [comment]
Okay I have a dirty workaround that at least allows us to still use vue-the-mask for the time being.

I do really like the syntax they use to create the mask directive. I think we should look at porting over the logic that does the actual masking.

**Comment by dev7ch** [comment]
ok, i ll have a look tomorrow and try another fix

**Comment by dev7ch** [comment]
or better maybe just fork https://github.com/vuejs-tips/vue-the-mask, merge the open PR fix of this issue and replace it with a new one when time is ready?

**Comment by rijkvanzanten** [comment]
Yeah we could keep our own fork of the thing, but at that point, we might as well clean it up and integrate it more closely into the codebase I suppose 🤷‍♂️

**Pull request merged**